src/Compilers/CSharp/Portable/Syntax/SyntaxFacts.cs

```diff
@@ -183,6 +183,9 @@ public static bool IsInTypeOnlyContext(ExpressionSyntax node)
                     case LocalFunctionStatement:
                         return ((LocalFunctionStatementSyntax)parent).ReturnType == node;
 
+                    case ParenthesizedLambdaExpression:
+                        return ((ParenthesizedLambdaExpressionSyntax)parent).ReturnType == node;
```

Comment:
What scenario did this affect?

Reply:
This is required to bind `dynamic` as an explicit return type since `BindNonGenericSimpleNamespaceOrTypeOrAliasSymbol()` requires `dynamic` to be in a type-only context.